src/Everscale/Address.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+#include "CellBuilder.h"
+#include "HexCoding.h"
+#include "Wallet.h"
+#include "WorkchainType.h"
+
+using namespace TW;
+using namespace TW::Everscale;
+
+bool Address::isValid(const std::string& string) {
+    auto parsed = parseWorkchainId(string);
+    if (!parsed.has_value())
+        return false;
+
+    auto [workchainId, pos] = *parsed;
+
+    if (workchainId != WorkchainType::Basechain && workchainId != WorkchainType::Masterchain) {
+        return false;
+    }
+
+    if (string.size() != pos + 64) {
+        return false;
+    }
+
+    std::string addr = string.substr(pos);
+    return parse_hex(addr).size() == size;
+}
+
+Address::Address(const std::string& string) {
+    if (!Address::isValid(string)) {
+        throw std::invalid_argument("Invalid address string!");
+    }
+
+    auto parsed = parseWorkchainId(string);
+    auto [wc, pos] = *parsed;
+
+    wc_ = wc;
```

Comment:
We recently discussed the coding style for the shadow variable. Can you rename all `field_` variables to `_field`